functions/Invoke-DbaDbShrink.ps1

```diff
@@ -0,0 +1,345 @@
+
+        .EXAMPLE
+            Invoke-DbaDatabaseShrink -SqlInstance sql2012 -AllUserDatabases
+
+            Shrinks all databases on SQL2012 (not ideal for production)
+
+    #>
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'Low')]
+    param (
+        [parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [Alias("ServerInstance", "SqlServer")]
+        [DbaInstanceParameter[]]$SqlInstance,
+        [Alias("Credential")]
+        [PSCredential]$SqlCredential,
+        [Alias("Databases")]
+        [object[]]$Database,
+        [object[]]$ExcludeDatabase,
+        [switch]$AllUserDatabases,
+        [ValidateRange(0, 99)]
+        [int]$PercentFreeSpace = 0,
+        [ValidateSet('Default', 'EmptyFile', 'NoTruncate', 'TruncateOnly')]
+        [string]$ShrinkMethod = "Default",
+        [ValidateSet('All', 'Data', 'Log')]
+        [string]$FileType = "All",
+        [int]$StepSizeMB,
+        [int]$StatementTimeout = 0,
+        [switch]$LogsOnly,
+        [switch]$ExcludeIndexStats,
+        [switch]$ExcludeUpdateUsage,
+        [Alias('Silent')]
+        [switch]$EnableException
+    )
+
+    begin {
+        if ($LogsOnly) {
+            Test-DbaDeprecation -DeprecatedOn "1.0.0" -Parameter "LogsOnly"
+            $FileType = 'Log'
+        }
+
+        $StatementTimeoutSeconds = $StatementTimeout * 60
+
+        $sql = "SELECT
+                  avg(avg_fragmentation_in_percent) as [avg_fragmentation_in_percent]
+                , max(avg_fragmentation_in_percent) as [max_fragmentation_in_percent]
+                FROM sys.dm_db_index_physical_stats (DB_ID(), NULL, NULL, NULL, NULL) AS indexstats
+                WHERE indexstats.avg_fragmentation_in_percent > 0 AND indexstats.page_count > 100
+                GROUP BY indexstats.database_id"
+    }
+
+    process {
+        if (!$Database -and !$ExcludeDatabase -and !$AllUserDatabases) {
+            Stop-Function -Message "You must specify databases to execute against using either -Database, -Exclude or -AllUserDatabases" -Continue
+        }
+
+        foreach ($instance in $SqlInstance) {
+            Write-Message -Level Verbose -Message "Connecting to $instance"
+            try {
+                $server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $SqlCredential
+            }
+            catch {
+                Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+            }
+
+            # changing statement timeout to $StatementTimeout
+            if ($StatementTimeout -eq 0) {
+                Write-Message -Level Verbose -Message "Changing statement timeout to infinity"
+            }
+            else {
+                Write-Message -Level Verbose -Message "Changing statement timeout to $StatementTimeout minutes"
+            }
+            $server.ConnectionContext.StatementTimeout = $StatementTimeoutSeconds
```

Comment:
I don't think this if/else block is really needed just for different verbose messages...I'd update this to just two statements, unless someone sees an issue:

```powershell
$server.ConnectionContext.StatementTimeout = $StatementTimeout
Write-Mesage -Level Verbose -Message "Connection timeout set to $StatementTimeout"
```
Just keep it simple since we have a default value set on the parameter.

Reply:
I would agree, thanks!